_posts/2019-01-02-Deviceid-Reliability.md

```diff
@@ -0,0 +1,95 @@
+
+The data can be described as a collection of deviceIDs which visit an app at a certain time.
+<p align="center"> <img src="../images/Deviceid-Reliability/table_data_example.png"> </p>
+
+
+Let's go into details of the discovered patterns:
+
+* **Fraction of unique devices**<br/>
+Only apps with a certain number of distinct deviceIDs must be considered to avoid statistical fluctuations.
+With the histogram of the fraction of unique devices we can see both apps which are visited by too many/few distinct deviceIDs as also apps which shows some regular traffic.
+![frac_uniq_dev](../images/Deviceid-Reliability/fraction_unique_devices.png)
+The peaks in at one are those which are visited by always distinct deviceid while the other agglomerate close to zero.
+
+We suspect that something is broken in the way the apps provide/access the deviceID information for those apps.
+All deviceIDs which appear in a non reliable flagged app are considered non reliable, a simple inner join over the selected apps is enought to create the blacklist.
+
+
+* **Non realistic number of apps**<br/>
+For a fixed amount of time (e.g. 1 day) it is possible to observe the number of apps visited by each deviceID.
+From this plot we found out some clear outliers which cannot correspond to real human users.
+![num_uniqu_apps](../images/Deviceid-Reliability/num_unique_apps.png)
+In the plot the number of the vistited apps is standardized (the absolute value minus the overall mean divided by the standard deviation), so that the presence of outliers is evident.
+
+
+* **Short period of time**:<br/>
+When we observe a deviceID on our data for a very short period of time (say 1 minute) but we do not register any other occurrence of the deviceID over several days both in the past and in the future, we do not consider these deviceIDs to be worth of further analysis.
+In fact, it is not sure that we have a real person behind it, for example a machine which automatically reset the value would produce the same data.
+![one_over_more_days](../images/Deviceid-Reliability/only_1_observations_over_days.png)
+The plot above represents the observation of a deviceID in a 5 days period, both for users which shown a regular activity (d6, d7, d8) and for those which appear only once (d1, d2, d3).
+The y axis is used to facilitate the graphical visualization of the activity patterns.
+
+
+* **Randomly generated deviceID**:<br/>
+Some app could provide a randomly generated deviceID.
+If this is the case, it is expected that the same deviceID is not observed in any other apps.<br/>
+We can think about two ways to generate the values:
+1) a new random deviceID is created for every acces;
+2) a particular transformation of the deviceID is used, thus the provided deviceID for the same real device is always the same but it will not be allined with the one provided by other apps.<br/>
+The query below allows us to catch these apps:
+```sql
```

Comment:
You don't need to give that query. Especially since you do not give any other query.